app/controllers/citizens/check_answers_controller.rb

```diff
@@ -9,6 +9,7 @@ def index
 
     def continue
       legal_aid_application.complete_means! unless legal_aid_application.means_completed?
+      legal_aid_application.update!(provider_step: :merits_start)
```

Comment:
This line and the new code in `providers_helper.rb` are kind of duplications.
They both say : 
`When the means are completed, the providers next step is the start of the merits flow`.
This logic shouldn't be in 2 places.

Actually it would make sense to update `provider_step` in the state machine:
```ruby
# app/models/concerns/legal_aid_application_state_machine.rb

      event :complete_means do
        transitions from: :checking_citizen_answers, to: :means_completed,
                    after: -> do
                      CleanupCapitalAttributes.call(self)
                      update!(provider_step: :client_received_legal_helps)
                    end
        transitions from: :checking_passported_answers, to: :means_completed
      end
```

or 

```ruby
      event :complete_means do
        transitions from: :checking_citizen_answers, to: :means_completed,
                    after: -> do
                      CleanupCapitalAttributes.call(self)
                      update!(provider_step: Flow::Flows::ProviderMerits::STEPS[:merits_start])
                    end
        transitions from: :checking_passported_answers, to: :means_completed
      end
```

Reply:
Thanks. Good idea and now it's implemented.